projects/angular-auth-oidc-client/src/lib/flows/callback-handling/state-validation-callback-handler.service.ts

```diff
@@ -21,21 +22,23 @@ export class StateValidationCallbackHandlerService {
   // STEP 4 All flows
 
   callbackStateValidation(callbackContext: CallbackContext, configId: string): Observable<CallbackContext> {
-    const validationResult = this.stateValidationService.getValidatedStateResult(callbackContext, configId);
-    callbackContext.validationResult = validationResult;
+    return this.stateValidationService.getValidatedStateResult(callbackContext, configId).pipe(
+      map((validationResult: StateValidationResult) => {
+        callbackContext.validationResult = validationResult;
 
-    if (validationResult.authResponseIsValid) {
-      this.authStateService.setAuthorizationData(validationResult.accessToken, callbackContext.authResult, configId);
+        if (validationResult.authResponseIsValid) {
+          this.authStateService.setAuthorizationData(validationResult.accessToken, callbackContext.authResult, configId);
 
-      return of(callbackContext);
-    } else {
-      const errorMessage = `authorizedCallback, token(s) validation failed, resetting. Hash: ${this.doc.location.hash}`;
-      this.loggerService.logWarning(configId, errorMessage);
-      this.resetAuthDataService.resetAuthorizationData(configId);
-      this.publishUnauthorizedState(callbackContext.validationResult, callbackContext.isRenewProcess);
-
-      return throwError(() => new Error(errorMessage));
-    }
+          return callbackContext;
+        } else {
+          const errorMessage = `authorizedCallback, token(s) validation failed, resetting. Hash: ${this.doc.location.hash}`;
+          this.loggerService.logWarning(configId, errorMessage);
+          this.resetAuthDataService.resetAuthorizationData(configId);
+          this.publishUnauthorizedState(callbackContext.validationResult, callbackContext.isRenewProcess);
+          throw new Error(errorMessage);
```

Comment:
Please use rxjs operator: return throwError(() => new Error(errorMessage));

Reply:
Using `return throwError(() => new Error(errorMessage));` would make the method return `Observable<CallbackContext | Observable<never>>` and cause errors in `FlowsService`, do you think it's the best approach? We would need some checks and rewrite in FlowsService.

Reply:
Nice catch. I skipped the `map` operator for some reason. Okay then, `throw new Error()` is okay then, just add a new line before.

Reply:
Done :)